File: family_tree_server/src/main/java/ru/cfif/cs/familytree/model/Person.java
package ru.cfif.cs.familytree.model;

import java.time.LocalDate;

public class Person {

	private final long id;
	private final long familyId;
	private final String name;
	private final String surname;
	private final LocalDate birthday;
	private final LocalDate deathday;

	public Person(long id, long familyId, String name, String surname, LocalDate birthday, LocalDate deathday) {
		this.id = id;
		this.familyId = familyId;
		this.name = name;
		this.surname = surname;
		this.birthday = birthday;
		this.deathday = deathday;
	}

	public Person(long id, Person person) {
		this.id = id;
		this.familyId = person.getFamilyId();
		this.name = person.getName();
		this.surname = person.getSurname();
		this.birthday = person.getBirthday();
		this.deathday = person.getDeathday();
	}

	public long getFamilyId() {
		return familyId;
	}

	public long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public String getSurname() {
		return surname;
	}

	public LocalDate getBirthday() {
		return birthday;
	}

	public LocalDate getDeathday() {
		return deathday;
	}
}

File: family_tree_view/src/main/resources/static/js/objects/Person.js
const FAKE_ID = -1;

class Person {
	constructor(name, surname, birthday, deathday, id) {
		this.id = id ? id : FAKE_ID;
		this.name = name;
		this.surname = surname;
		this.birthday = birthday === null ? ' ... ' : birthday;
		this.deathday = deathday === null ? ' ... ' : deathday;
	}
	
	getFullName() {
		return this.name + ' ' + this.surname;
	}
}

export default Person;

export function clonePerson(person) {
	return new Person(person.name, person.surname, person.birthday, person.deathday, person.id);
}